src/api/handlers/appointments/postAppointments.js

```diff
@@ -24,9 +28,52 @@ export default (req, res) => {
     appointmentIsConfirmed: false
   };
 
-  db("appointments")
-  .insert(appointment)
-  .then(result => {
-    res.status(201).send(result);
-  });
+  let insertedAppointmentId;
+  try {
+    insertedAppointmentId = await db("appointments").insert(appointment);
+  } catch (error) {
+    res.status(500).send({
+      error,
+      message: `Could not insert appointment for ${patientPhoneNumber}`
+    });
+    return;
+  }
+
+  const language = LanguagesModel.getByLanguageString(patientLanguage);
+  const template = TemplatesModel.getById(1);
+  const messageBody = TemplatesModel.generateMessage(1, language.id, appointment);
+
+  const message = {
+    appointmentId: insertedAppointmentId,
+    messageBody,
+    language: language.id,
+    templateName: template.templateName,
+    // UTC
+    timeSent: moment().format("YYYY-MM-DD HH:mm:ss")
+  };
+
+  sendMessage(patientPhoneNumber, messageBody)
+    .then(async () => {
+      let messageId;
+      try {
+        messageId = await db("messages").insert(message);
+      } catch (error) {
+        res.status(500).send({
```

Comment:
If you aren't returning here does it continue execution and try and do the res.status(200)?

Reply:
whoops. yup.

Reply:
updated.

Reply:
Is this better:
```
  await sendMessage(patientPhoneNumber, messageBody)
    .catch((error) => {
      res.status(500).send({ error, message: "Failed to send appointment reminder"});
      return;
    });

  let messageId;
  try {
    messageId = await db("messages").insert(message);
  } catch (error) {
    res.status(500).send({
      error,
      message: "Could not save message to database"
    });
    return;
  }
  res.status(200).send({
    insertedAppointmentId,
    messageId: messageId[0]
  });
  return;
```